src/include/hip/rpp_hip_common.hpp

```diff
@@ -1481,4 +1618,420 @@ __device__ __forceinline__ void rpp_hip_math_subtract24_const(d_float24 *src_f24
+{
+    d_float6_as_float3s src_f6;
+    int srcIdx = (int)locSrcFloor->y * srcStrideH + (int)locSrcFloor->x * 3;
+    src_f6 = *(d_float6_as_float3s *)&srcPtr[srcIdx];
+    srcNeighborhood_f12->x.x = src_f6.x.x;
+    srcNeighborhood_f12->x.y = src_f6.y.x;
+    srcNeighborhood_f12->y.x = src_f6.x.y;
+    srcNeighborhood_f12->y.y = src_f6.y.y;
+    srcNeighborhood_f12->z.x = src_f6.x.z;
+    srcNeighborhood_f12->z.y = src_f6.y.z;
+    srcIdx += srcStrideH;
+    src_f6 = *(d_float6_as_float3s *)&srcPtr[srcIdx];
+    srcNeighborhood_f12->x.z = src_f6.x.x;
+    srcNeighborhood_f12->x.w = src_f6.y.x;
+    srcNeighborhood_f12->y.z = src_f6.x.y;
+    srcNeighborhood_f12->y.w = src_f6.y.y;
+    srcNeighborhood_f12->z.z = src_f6.x.z;
+    srcNeighborhood_f12->z.w = src_f6.y.z;
+}
+
+// I8 loads for bilinear interpolation (12 I8 pixels)
+
+__device__ __forceinline__ void rpp_hip_interpolate3_bilinear_load_pkd3(schar *srcPtr, uint srcStrideH, float2 *locSrcFloor, d_float12 *srcNeighborhood_f12)
+{
+    int2 src_i2;
+    int srcIdx = (int)locSrcFloor->y * srcStrideH + (int)locSrcFloor->x * 3;
+    src_i2 = *(int2 *)&srcPtr[srcIdx];
+    srcNeighborhood_f12->x.x = rpp_hip_unpack0(src_i2.x);
+    srcNeighborhood_f12->x.y = rpp_hip_unpack3(src_i2.x);
+    srcNeighborhood_f12->y.x = rpp_hip_unpack1(src_i2.x);
+    srcNeighborhood_f12->y.y = rpp_hip_unpack0(src_i2.y);
+    srcNeighborhood_f12->z.x = rpp_hip_unpack2(src_i2.x);
+    srcNeighborhood_f12->z.y = rpp_hip_unpack1(src_i2.y);
+    srcIdx += srcStrideH;
+    src_i2 = *(int2 *)&srcPtr[srcIdx];
+    srcNeighborhood_f12->x.z = rpp_hip_unpack0(src_i2.x);
+    srcNeighborhood_f12->x.w = rpp_hip_unpack3(src_i2.x);
+    srcNeighborhood_f12->y.z = rpp_hip_unpack1(src_i2.x);
+    srcNeighborhood_f12->y.w = rpp_hip_unpack0(src_i2.y);
+    srcNeighborhood_f12->z.z = rpp_hip_unpack2(src_i2.x);
+    srcNeighborhood_f12->z.w = rpp_hip_unpack1(src_i2.y);
+}
+
+// F16 loads for bilinear interpolation (12 F16 pixels)
+
+__device__ __forceinline__ void rpp_hip_interpolate3_bilinear_load_pkd3(half *srcPtr, uint srcStrideH, float2 *locSrcFloor, d_float12 *srcNeighborhood_f12)
+{
+    d_half6 src_h6;
+    d_float6 src_f6;
+    d_float6_as_float3s *src_f6f3s = (d_float6_as_float3s *)&src_f6;
+    int srcIdx = (int)locSrcFloor->y * srcStrideH + (int)locSrcFloor->x * 3;
+    src_h6 = *(d_half6 *)&srcPtr[srcIdx];
+    src_f6.x = __half22float2(src_h6.x);
+    src_f6.y = __half22float2(src_h6.y);
+    src_f6.z = __half22float2(src_h6.z);
+    srcNeighborhood_f12->x.x = src_f6f3s->x.x;
+    srcNeighborhood_f12->x.y = src_f6f3s->y.x;
+    srcNeighborhood_f12->y.x = src_f6f3s->x.y;
+    srcNeighborhood_f12->y.y = src_f6f3s->y.y;
+    srcNeighborhood_f12->z.x = src_f6f3s->x.z;
+    srcNeighborhood_f12->z.y = src_f6f3s->y.z;
+    srcIdx += srcStrideH;
+    src_h6 = *(d_half6 *)&srcPtr[srcIdx];
+    src_f6.x = __half22float2(src_h6.x);
+    src_f6.y = __half22float2(src_h6.y);
```

Comment:
change to what we have discussed

Reply:
Added the fix for this issue. Will be checking on the xyz notation changes in a separate pr.